builder/utils/deployer.py

```diff
@@ -398,35 +399,57 @@ def deploy_model(self, image_ecr_path, model_s3_path):
             )
             return None
 
-    def deploy(self, s3_uri):
+    def deploy(self, s3_uri, endpoint_only=False):
         deployed, delayed, ex_msg = False, False, ""
         try:
-            self.delete_existing_endpoints()
-            os.chdir(self.root_dir)
-            setup_config_handler, setup_config, s3_dir = self.load_model(s3_uri)
-            image_ecr_path = self.build_and_push_docker(
-                setup_config_handler, setup_config
-            )
-            model_s3_path = self.archive_and_upload_model(setup_config, s3_dir)
-            endpoint_url = self.deploy_model(image_ecr_path, model_s3_path)
+            if not endpoint_only:
+                self.delete_existing_endpoints()
+                os.chdir(self.root_dir)
+                setup_config_handler, setup_config, s3_dir = self.load_model(s3_uri)
+                image_ecr_path = self.build_and_push_docker(
+                    setup_config_handler, setup_config
+                )
+                model_s3_path = self.archive_and_upload_model(setup_config, s3_dir)
+                endpoint_url = self.deploy_model(image_ecr_path, model_s3_path)
+            else:
+                s3_bucket, s3_path, s3_dir, endpoint_name = self.parse_s3_uri(s3_uri)
```

Comment:
This else block ignores creating the image/ecr, and just creates the model + endpoint + endpoint config